admin/lib/validator/school-impersonation-dfe-number-validator.js

```diff
@@ -0,0 +1,22 @@
+const ValidationError = require('../validation-error')
+const SchoolImpersonationMessages = require('../errors/school-impersonation')
+
+const isDfeNumberEmpty = (dfeNumber) => {
+  let validationError = new ValidationError()
+  if (!dfeNumber) {
+    validationError.addError('dfeNumber', SchoolImpersonationMessages.noInput)
+    return validationError
+  }
+  return validationError
+}
+
+const isDfeNumberValid = (school) => {
+  let validationError = new ValidationError()
+  if (!school || !school.id) {
```

Comment:
`{
  id: 'foo'
}`

passes validation but is not valid.

Reply:
@jon-shipley at that stage the school either has been fetched from the database or it doesn't exist at all.if a 'foo' value is stored as id that would be fundamentally wrong.